### ISSUE TITLE
[Feature?]: Wrangler/Miniflare dev server for Cloudflare

### ISSUE THREAD

**Issue by rvlzzr** [solidjs/solid-start#1394]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Solid-start's dev server doesn't use wrangler or miniflare for the Cloudflare presets, so bindings aren't available etc and it isn't really usable.

There's a hacky example in vinxi of react ssr with a miniflare 2 dev server, but miniflare 2 is 3 years out-of-date and is missing a lot of things, to even get D1 to work you have to prefix the binding with` __D1_BETA__` etc.

### Examples 🌈

Can the dev server under Cloudflare use https://developers.cloudflare.com/workers/wrangler/api/#getplatformproxy perhaps?

Sveltekit is doing this: https://github.com/sveltejs/kit/pull/11732/files#diff-78806028d48ad47894445421562b7796c53178948c017e13d4550def4193903e

### Motivation 🔦

Solid-start DX with Cloudflare is currently miserable.

**Comment by rvlzzr** [comment]
It seems you can just do something like the following in middleware, but still I think this is something that should be handled by the framework so process.env works consistently in dev/prod, or at least documented.

```typescript
if (import.meta.env.PROD) {
  event.locals.db = process.env.DB;
} else {
  const wrangler = await import("wrangler");
  const proxy = await wrangler.getPlatformProxy();
  event.locals.db = proxy.env.DB;
}
```

**Comment by ryansolid** [comment]
Yeah since we don't control the adapters/presets this is a bit more tricky then in the past (or in Sveltekit). It was one of the tradeoffs of going this way, but I hope to have a solution in the future.

**Comment by mattp0123** [comment]
There's an [Access to the bindings in local env](https://nitro.unjs.io/deploy/providers/cloudflare#access-to-the-bindings-in-local-env) section in Nitro docs. But it doesn't seem to work with Solid Start, according to https://github.com/pi0/nitro-cloudflare-dev/issues/42

**Comment by ryansolid** [comment]
Yeah Nitro added some newer features to support this. We don't use Nitro currently in dev with Vinxi so this won't work. I'm hoping after Vite 6 core Vite(and in so Nitro) will have the features we need to change up this setup.

**Comment by bilner** [comment]
If it's useful to others coming here, I used bun patch to alter `node_modules/vinxi/lib/nitro-dev.js` to add middleware to add the result of `getPlatformProxy` to the event context instead. Now my application code is exactly the same in dev as well as when using wrangler/deployed on Cloudflare.

**Comment by huseeiin** [comment]
there are 2 possible ways this could be fixed:
1. we get nitro in dev support and then we can just use nitro-cloudflare-dev
2. solidstart moves from vinxi to vite's environments api and then we can just use @cloudflare/vite-plugin

**Comment by birkskyum** [comment]
Closed as obsoleted, because there's now a [Cloudflare vite plugin](https://www.npmjs.com/package/@cloudflare/vite-plugin) which can be used with SolidStart v2
